action_server_bt/src/action_server_bt.cpp

```diff
@@ -119,30 +124,33 @@ void ActionServerBT::execute(const std::shared_ptr<GoalHandleActionTree> goal_ha
       feedback->node_status = convert_node_status(status);
       goal_handle->publish_feedback(feedback);
 
-      if (!loopRate.sleep())
-      {
-        RCLCPP_WARN(kLogger, "Behavior Tree tick frequency %d Hz was exceeded!",
-                    static_cast<int>(1.0 / (loopRate.period().count() * 1.0e-9)));
-      }
+      // sleep to tick the tree at the desired frequency
+      tree.sleep(std::chrono::milliseconds(static_cast<int>(1000.0 / params_.behavior_tick_frequency)));
```

Comment:
To keep the rate constant, we should do something like this:
```c++

const auto period = std::chrono::milliseconds(static_cast<int>(1000.0 / params_.behavior_tick_frequency));
auto next_deadline = std::chrono::steady_clock::now() + period;

// the wile loop
while(...)
{
    // your code (execute)
    auto now = std::chrono::steady_clock::now();
    if(now < next_deadline) 
    {
      tree.sleep(deadline - now);
    }
    deadline += period;
}
```

Reply:
Thanks for adding this!